Update gcc/g++ settings for PAX on RockyLinux 8

tuhaihe opened the pull request:
Cloudberry can now be built with PAX support using the default GCC/G++
toolchain on Rocky Linux 8. There is no longer a need to upgrade
GCC/G++.

The default GCC/G++ version on Rocky Linux 8 are:
 - gcc 8.5.0
 - g++ 8.5.0

See PR: https://github.com/apache/cloudberry/pull/1165

tuhaihe commented:
Once #1165 is merged, we can merge this PR.

tuhaihe commented:
Can build the Cloudberry on the latest Docker dev with this commit. It's ready to be merged now.

edespino commented:
@tuhaihe Can you provide the default GCC version on Rocky Linux 8? I want to ensure I am using the correct compiler.

tuhaihe commented:
> @tuhaihe Can you provide the default GCC version on Rocky Linux 8? I want to ensure I am using the correct compiler.

Hi @edespino, sure. Here are the details:

```bash
[gpadmin@cdw cloudberry]$ gcc --version
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-26)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ g++ --version
g++ (GCC) 8.5.0 20210514 (Red Hat 8.5.0-26)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

edespino commented on images/docker/cbdb/build/rocky9/Dockerfile:
```diff
@@ -138,7 +138,7 @@ RUN dnf makecache && \
         perl-IPC-Run \
         protobuf-devel && \
-    dnf clean all && \
+    dnf clean all && rm -rf /var/cache/dnf/* && \
```
Maybe amend the commit message with the following which describes this change.

Purpose of rm -rf /var/cache/dnf/*

Even though dnf clean all should remove the contents of /var/cache/dnf/, this extra rm step is:

* A failsafe/manual cleanup, in case DNF leaves anything behind (e.g. due to bugs or distro version differences).
* Common in slim or hardened images to ensure absolutely no cache data remains.

tuhaihe replied:
Good. Let me leave it in the next single PR.

edespino commented on images/docker/cbdb/build/rocky8/Dockerfile:
```diff
@@ -87,8 +87,8 @@ RUN dnf makecache && \
         flex \
-        gcc-toolset-11-gcc \
-        gcc-toolset-11-gcc-c++ \
+        gcc \
```
Please consider amending the commit message to include the default GCC version on Rocky Linux 8. I believe GCC 8.5.0 is used:

```
$ rpm -aq | grep gcc
gcc-8.5.0-26.el8_10.x86_64
gcc-c++-8.5.0-26.el8_10.x86_64
gcc-plugin-annobin-8.5.0-26.el8_10.x86_64
libgcc-8.5.0-26.el8_10.x86_64
```

tuhaihe replied:
Thanks. Updated.

edespino reviewed (dismissed):
Please consider the Commit message description amends recommendations. I was able to build without issue on Rocky Linux 8 AWS EC2 VM environment.

tuhaihe committed:
Update gcc/g++ settings for PAX on RockyLinux 8

Cloudberry can now be built with PAX support using the default GCC/G++
toolchain on Rocky Linux 8. There is no longer a need to upgrade
GCC/G++.

The default GCC/G++ version on Rocky Linux 8 are:
 - gcc 8.5.0
 - g++ 8.5.0

See PR: https://github.com/apache/cloudberry/pull/1165

TomShawn reviewed (approved):
LGTM
